app/controllers/admin/coupons_controller.rb

```diff
@@ -1,10 +1,19 @@
 module Admin
   class CouponsController < AdminController
 
+    def index
```

Comment:
you should be able to put this whole index function in the concern. 

also you have to include your concern in here.